py-polars/polars/lazyframe/frame.py

```diff
@@ -4074,13 +4074,13 @@ def join(
                 Returns all rows from the right table, and the matched rows from the
                 left table
             * *full*
-                 Returns all rows when there is a match in either left or right table
+                Returns all rows when there is a match in either left or right table
             * *cross*
-                 Returns the Cartesian product of rows from both tables
+                Returns the Cartesian product of rows from both tables
             * *semi*
-                 Filter rows that have a match in the right table.
+                Return rows of only the left table, that have a match in the right table.
             * *anti*
-                 Filter rows that not have a match in the right table.
+                Return rows of only the left table, that have no match in the right table.
```

Comment:
```suggestion
                Returns rows from the left table that have no match in the right table.
```

Reply:
Can we make it more explicit that _only_ the left table is returned? Because that was the surprising part for me, since I read the documentation with the expectation that two tabels are joined, i.e. always both tables are returned.

Reply:
> Can we make it more explicit that _only_ the left table is returned? Because that was the surprising part for me, since I read the documentation with the expectation that two tabels are joined, i.e. always both tables are returned.

This is as explicit as necessary: "from the left table"; If the rows are coming from the left table, they _cannot_ be coming from the right table ;)